### DISCUSSION TITLE
Request-Reply for Kafka - header.ReplyTo is not doing anything

### DISCUSSION THREAD

**Discussion by hooi5683** [BrighterCommand/Brighter#2403] [General]
I've took RMQRequestReply project with some modification to test Request-Reply behind kafka. below are some of my findings I wish to get clarification.

1. Reply topics generated by the producer are the corelationId instead of the name specified in MessageHeader.ReplyTo
2. MessageHeader.ReplyTo is not being serialized into the message header, thus Consumer only referring to corelationId as the topic name for the reply

Is there a way I can just maintain 1 reply topic instead of many reply topics going to be created on the fly?

**Comment by iancooper** [comment]
Hi @hooi5683 I would need to look at what we need for Request-Reply with Kafka as that feature is mainly oriented around queue based solutions not stream based solutions. We sit on the fence around features for layering 'queue features' like Requeue with Delay and DLQ over Kafka. We will get to them, but because you can just implement them yourself (request reply is just send command, and listen for event with correlation id) they are a lower priority

**Reply by hooi5683** [reply]
Thank you @iancooper.
regarding message.header.ReplyTo is not published in the message header, shall the method below to be improved?
https://github.com/BrighterCommand/Brighter/blob/e95a60eb1c8513f324674002e30d502e5145c5b7/src/Paramore.Brighter.MessagingGateway.Kafka/KafkaMessagePublisher.cs#L41

**Reply by iancooper** [reply]
That's the area, but we can't simply default everything to camel - there will be inflight messages etc. - we would need to make it a publication option and pass it down

**Reply by hooi5683** [reply]
should header.ReplyTo to be included into message header when publish call PublishMessageAsync() @ src\Paramore.Brighter.MessagingGateway.Kafka\KafkaMessagePublisher.cs  ?  

not sure is this a bug in BuildMessage()

**Reply by iancooper** [reply]
@hooi5683 It is one of our standard headers, so possibly it should be added.

Feel free to raise on the issue around camel casing as someone is in that code looking at supporting that
